test/Cli/Func.TestFramework/Commands/FuncCommand.cs

```diff
@@ -0,0 +1,231 @@
+                {
+                    AutoFlush = true
+                };
+
+                // Write initial information
+                FileWriter.WriteLine($"=== Test started at {DateTime.Now} ===");
+                FileWriter.WriteLine($"Test Name: {spec.TestName}");
+                var display = $"func {string.Join(" ", spec.Arguments)}";
+                FileWriter.WriteLine($"Command: {display}");
+                FileWriter.WriteLine($"Working Directory: {spec.WorkingDirectory ?? "not specified"}");
+                FileWriter.WriteLine("====================================");
+
+                command.OnOutputLine(line =>
+                {
+                    try
+                    {
+                        // Write to the file if it's still open
+                        if (FileWriter != null && FileWriter.BaseStream != null)
+                        {
+                            FileWriter.WriteLine($"[STDOUT] {line}");
+                            FileWriter.Flush();
+                        }
+
+                        Log.WriteLine($"》   {line}");
+                        CommandOutputHandler?.Invoke(line);
+                    }
+                    catch (Exception ex)
+                    {
+                        Log.WriteLine($"Error writing to log file: {ex.Message}");
+                    }
+                });
+
+                command.OnErrorLine(line =>
+                {
+                    try
+                    {
+                        // Write to the file if it's still open
+                        if (FileWriter != null && FileWriter.BaseStream != null)
+                        {
+                            FileWriter.WriteLine($"[STDERR] {line}");
+                            FileWriter.Flush();
+                        }
+
+                        if (!string.IsNullOrEmpty(line))
+                        {
+                            Log.WriteLine($"❌   {line}");
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Log.WriteLine($"Error writing to log file: {ex.Message}");
+                    }
+                });
+
+                Log.WriteLine($"Executing '{display}':");
+                Log.WriteLine($"Output being captured to: {LogFilePath}");
+
+                var result = ((Command)command).Execute(ProcessStartedHandler, FileWriter);
+
+                FileWriter.WriteLine("====================================");
+                FileWriter.WriteLine($"Command exited with code: {result.ExitCode}");
+                FileWriter.WriteLine($"=== Test ended at {DateTime.Now} ===");
+
+                Log.WriteLine($"Command '{display}' exited with exit code {result.ExitCode}.");
+
+                return result;
+            }
+            finally
+            {
+                // Make sure to close and dispose the writer
+                if (FileWriter != null)
```

Comment:
nit: apply to most instance of !=

```suggestion
                if (FileWriter is not null)
```